apps/spruce/src/pages/waterfall/caching/index.ts

```diff
@@ -10,7 +10,7 @@ export const readVersions = ((existing, { args, readField }) => {
   const minOrder = args?.options?.minOrder ?? 0;
   const maxOrder = args?.options?.maxOrder ?? 0;
   const limit = args?.options?.limit ?? VERSION_LIMIT;
-  const { mostRecentVersionOrder = 0 } =
+  const { activeVersionIds = [], mostRecentVersionOrder = 0 } =
```

Comment:
Oof, something I noticed (which we don't have to fix for the beta test, since it sounds hard) is that I think the cache also needs to check the build variant match. Otherwise, it just returns 5 activated versions, but some of those could be collapsed due to the filter, so you have less than 5 matching versions on the page